includes/Plugin.php

```diff
@@ -87,6 +87,29 @@ function() {
 			return;
 		}
 
+		// In the case of IDNs, ensure the ASCII and non-ASCII domains
+		// are treated as allowable origins.
+		add_filter(
+			'allowed_redirect_hosts',
+			function( $hosts ) {
+				$wpp = wp_parse_url( home_url() );
+
+				// If this host isn't an IDN (or is already an ASCII-formatted
+				// IDN), we can return the existing `$hosts` array unmodified.
+				if ( idn_to_ascii( $wpp['host'], 0, INTL_IDNA_VARIANT_UTS46 ) === $wpp['host'] ) {
```

Comment:
We can't rely on `idn_to_*` functions because these are contingent on the `intl` PHP extension. See https://github.com/google/site-kit-wp/issues/3606#issuecomment-875371118 for more context.

Reply:
Oh, shoot, I didn't realise that one. Alright, I'll take a look at this one again to sort it out. 👍🏻

Reply:
Turns out a check for it being an ASCII string is sufficient here, which makes sense.

I've updated it to use an `mb_*` function instead, since we're using them elsewhere in the plugin already so should be safe. Should be sorted now 👍🏻